src/control/lib/control/rpc.go

```diff
@@ -208,6 +208,9 @@ func (c *Client) InvokeUnaryRPCAsync(parent context.Context, req UnaryRequest) (
 					if err == nil {
 						msg, err = req.getRPC()(ctx, conn)
 					}
+					if conn != nil {
+						conn.Close()
+					}
```

Comment:
Is it possible that err != nil and conn != nil? If not, might be better to put the conn.Close in the if block above

Reply:
Good point. I have a follow-up PR that I'll roll that into.